src/api/app/models/package.rb

```diff
@@ -141,14 +141,9 @@ def self.check_cache(project, package, opts)
   end
 
   def self.internal_get_project(project)
-    if project.is_a?(Project)
-      prj = project
-    else
-      return if Project.is_remote_project?(project)
-      prj = Project.get_by_name(project)
-    end
-    raise UnknownObjectError, "#{project}/#{package}" unless prj
-    prj
+    return project if project.is_a?(Project)
+    return if Project.is_remote_project?(project)
+    Project.get_by_name(project)
```

Comment:
`Project.get_by_name` returns already a ` UnknownObjectError` if nothing is found. So this custom exception was never raised